recipes/gtsam/all/conanfile.py

```diff
@@ -120,6 +122,20 @@ def _patch_sources(self): #Needed to build GTSam as a subproject using the conan
                                   "/MDd ",
                                   "/{} ".format(self.settings.compiler.runtime))
 
+    def validate(self):
```

Comment:
We may need to check for the `header_only` option here. I'm not sure if that option remove all this modules or if it provides a headers only version of them.

I don't particularly like all the if statements one after the other, but I tried to iterate with something like
```
if self.options["boost"]."without_{}".format("component"):
```
But it doesn't work.

Reply:
I did that for folly:

```python
    @property
    def _required_boost_components(self):
        return ["context", "filesystem", "program_options", "regex", "system", "thread"]

    def validate(self):
        miss_boost_required_comp = any(getattr(self.options["boost"], "without_{}".format(boost_comp), True) for boost_comp in self._required_boost_components)
        if self.options["boost"].header_only or miss_boost_required_comp:
            raise ConanInvalidConfiguration("Folly requires these boost components: {}".format(", ".join(self._required_boost_components)))
```

Reply:
`getattr()` is the part that I was missing tok do this. Thanks!